File: backend/api/src/main/scala/com/github/ai/split/api/response/PostExpenseResponse.scala
package com.github.ai.split.api.response

import com.github.ai.split.api.ExpenseDto
import zio.json.{DeriveJsonEncoder, JsonEncoder}

case class PostExpenseResponse(
  expense: ExpenseDto
)

object PostExpenseResponse {
  implicit val encoder: JsonEncoder[PostExpenseResponse] = DeriveJsonEncoder.gen[PostExpenseResponse]
}

File: backend/api/src/main/java/com/github/ai/split/api/response/PostExpenseResponse.java
package com.github.ai.split.api.response;

import com.github.ai.split.api.ExpenseDto;

public class PostExpenseResponse {

    public ExpenseDto expense = new ExpenseDto();

    public PostExpenseResponse() {}

    public PostExpenseResponse(ExpenseDto expense) {
        this.expense = expense != null ? expense : new ExpenseDto();
    }
}